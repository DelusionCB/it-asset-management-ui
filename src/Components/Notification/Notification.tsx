import React, {useContext, useEffect} from 'react';
import {NotificationContext} from '../../Context/NotificationContext';
import {useTranslation} from 'react-i18next';
import './index.scss'

const Notification = (): JSX.Element => {
    const {hideNotification, show, message} = useContext(NotificationContext);
    const {t} = useTranslation()

    useEffect(() => {
        // Hide the notification after 5 seconds (5000ms)
        const timer = setTimeout(() => {
            hideNotification();
        }, 5000);

        return () => {
            clearTimeout(timer);
        };
    }, []);

    function getIcon (message: string): JSX.Element {
        let icon = ''
        switch (message) {
            case 'status.save':
            case 'status.create':
                icon = 'check-circle-fill'
                break;
            case 'status.delete':
                icon = 'trash3-fill'
                break;
            case 'status.error':
                icon = 'bug-fill'
                break;
            case 'status.authorization':
                icon = 'person-fill-exclamation'
                break;
        }
        return (<i className={`bi bi-${icon}`} />)
    }
    return (
        <>
            {show &&
                <dialog open={show} className='notification'>
                    <div>
                        {getIcon(message)}
                        <h1 role='status' className='notification-message'>{t(message)}</h1>
                    </div>
                </dialog>
            }
        </>
    );
};

export default Notification;
