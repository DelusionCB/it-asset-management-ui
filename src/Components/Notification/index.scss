.notification {
    position: fixed;
    bottom: 50%;
    transform: translateY(50%);
    width: 50%;
    height: 50%;
    z-index: 9999;
    background-color: var(--primary-color); /* Adjust the transparency as needed */
    opacity: 0.95;
    color: #fff;
    padding: 10px;
    border-radius: 20px;
    display: flex;
    justify-content: center;
    align-items: center;
    text-align: center;

    @keyframes scaleFont {
        0% {
            font-size: 0;
        }
        100% {
            font-size: 8rem;
        }
    }

    i {
        &::before {
            font-size: 0;
            color: white;
            animation: scaleFont 1.5s ease-in-out forwards;
        }
    }

    .notification-message {
        margin: 0;
        padding: 10px;
        text-align: center;
        font-size: 2rem;
    }
}
