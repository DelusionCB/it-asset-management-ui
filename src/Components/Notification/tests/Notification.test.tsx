import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import '@testing-library/jest-dom'
import {BrowserRouter} from 'react-router-dom';
import Notification from '../Notification';

describe('<Notification />', () => {
    let renderer: TestRenderer.ReactTestRenderer;

    // Render the component before tests
    beforeAll(() => {
        renderer = TestRenderer.create(
            <BrowserRouter>
                <Notification />
            </BrowserRouter>,
        );
    });
    describe('Snapshot', () => {
        it('renders correctly', () => {
            expect(renderer.toJSON()).toMatchSnapshot();
        })
    })
})
