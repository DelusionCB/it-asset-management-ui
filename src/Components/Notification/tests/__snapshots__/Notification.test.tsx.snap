// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notification /> Snapshot renders correctly 1`] = `null`;
